addons/placement/functions/fnc_updatePreview.sqf

```diff
@@ -0,0 +1,62 @@
+#include "script_component.hpp"
+
+private _pos = AGLtoASL screenToWorld getMousePosition;
+private _class = call FUNC(getSelectedClass);
+
+#define OFFSET 10
+
+if (_class isEqualTo "") exitWith {
+	if !(GVAR(previewVehicle) isEqualTo objNull) then {
+		deleteVehicle GVAR(previewVehicle);
+		deleteVehicle GVAR(previewObject);
+		GVAR(previewVehicle) = objNull;
+	};
+};
+
+// Delete the old preview if the player selects a different item
+if !(_class isEqualTo (typeOf GVAR(previewVehicle))) then {
+	deleteVehicle GVAR(previewVehicle);
+	deleteVehicle GVAR(previewObject);
+	GVAR(previewVehicle) = objNull;
+};
+
+// Create the preview if it doesn't exist
+if (isNull GVAR(previewVehicle)) then {
+    GVAR(previewObject) = "Land_Wrench_F" createVehicleLocal (ASLtoAGL _pos);
+	GVAR(previewObject) hideObject true;
+	GVAR(previewVehicle) = _class createVehicleLocal [0, 0, 0];
+	GVAR(previewVehicle) attachTo [GVAR(previewObject), [0, 0, OFFSET + 1 + ((boundingCenter GVAR(previewVehicle)) select 2)]];
+	GVAR(previewVehicle) enableSimulationGlobal false;
```

Comment:
Use createSimpleObject with local argument instead (next update, another PR)?

Reply:
This is the plan once that hits the main branch

Reply:
It will be in Contact DLC update version 1.94 on the 27th.